### PR TITLE
fix: retry on download errors

### PR DESCRIPTION
**Pull request by carlossg**
to avoid failing because of issues like connection reset

### CODE REVIEW

**Comment by yannh** [comment]
Good idea, though this never gives up and doesnt back off incrementallly.. Any library we might be able to use maybe? :thinking:

**Comment by carlossg** [comment]
It just tries once, one second later, but I've used now hashicorp/go-retryablehttp. 
If this looks good the tests need to use a real local server to test the retries

**Comment by carlossg** [comment]
@yannh added a test with a real http server to test the retries

File: pkg/registry/http_test.go [hunk 1]
```diff
@@ -3,114 +3,188 @@ package registry
...
+			[]byte("Normal response"),
 			nil,
 		},
 	} {
-		reg := SchemaRegistry{
-			c:                  testCase.c,
-			schemaPathTemplate: testCase.schemaPathTemplate,
-			strict:             testCase.strict,
+		// create a temporary directory for the cache
+		tmpDir, err := os.MkdirTemp("", "kubeconform-cache")
```
Comment (yannh) [inline pkg/registry/http_test.go:161]: Not sure about creating a cache folder here... Esp as we test the HTTP client?

**Comment by yannh** [comment]
Hi @carlossg , thanks a lot for the contribution. Your tests are quite a bit more complex than what we had before, but they look good. I iterated on them a little bit, can you let me know what you think? Thanks!

**Comment by carlossg** [comment]
much better thanks

**Comment by yannh** [comment]
Acceptance tests are failing, not sure why - they run locally just fine. Unsure if related to this PR :thinking: it ll take me some time to troubleshoot.

**Comment by yannh** [comment]
I reverted it, it seems to print debug log in some cases. I ll review this later today/this week :bow:

**Pull request merged**